### PR TITLE
GTEST/UCS: Updated ptr_map gtests.

### PR DESCRIPTION
**Pull request by rakhmets**
## What
Replaced multiple call for new operator by initializing vector.

## Why ?
Addressed
https://github.com/openucx/ucx/pull/7159#discussion_r689863284
https://github.com/openucx/ucx/pull/7159#issuecomment-901674724

### CODE REVIEW

File: test/gtest/ucs/test_datatype.cc [hunk 1]
```diff
@@ -1153,22 +1151,30 @@ UCS_TEST_F(test_array, fixed_onstack) {
...
-    UCS_PTR_MAP_T(test_st) ptr_map;
+UCS_TEST_F(test_datatype_ptr_map, unsafe_put) {
+    UCS_PTR_MAP_T(unsafe_put) ptr_map;
 
-    const size_t N = 10 * UCS_KBYTE;
-    ucs_ptr_map_key_t ptr_key;
-    std_map_t         std_map;
-    ucs_status_t      status;
+    std_map_t std_map;
+    std_vec_t std_vec(10 * UCS_KBYTE, 0);
```
Comment (evgeny-leksikov) [inline test/gtest/ucs/test_datatype.cc:1169]: it's confusing to use KBYTE for int elem number (actually array size is 40KB), better to use `10000` or `10*1024` or `1 << 10`

File: test/gtest/ucs/test_datatype.cc [hunk 2]
```diff
@@ -1160,13 +1160,16 @@ class test_datatype_ptr_map : public test_datatype {
     typedef std_map_t::const_iterator std_map_it_t;
     typedef std::vector<int> std_vec_t;
     typedef std_vec_t::iterator std_vec_it_t;
+
+    static const size_t N             = 10 * 1 << 10;
```
Comment (evgeny-leksikov) [inline test/gtest/ucs/test_datatype.cc:1164]: pls add ()

File: test/gtest/ucs/test_datatype.cc [hunk 3]
```diff
@@ -1160,13 +1160,16 @@ class test_datatype_ptr_map : public test_datatype {
...
+    static const unsigned num_threads = 4;
```
Comment (evgeny-leksikov) [inline test/gtest/ucs/test_datatype.cc:1165]: 👍

File: test/gtest/ucs/test_datatype.cc [hunk 4]
```diff
@@ -1153,22 +1151,33 @@ UCS_TEST_F(test_array, fixed_onstack) {
...
+    UCS_PTR_MAP_DEFINE(safe_put, 1);
+
+    typedef std::map<ucs_ptr_map_key_t, int*> std_map_t;
+    typedef std_map_t::const_iterator std_map_it_t;
+    typedef std::vector<int> std_vec_t;
+    typedef std_vec_t::iterator std_vec_it_t;
 
-UCS_TEST_F(test_datatype, ptr_map) {
-    UCS_PTR_MAP_T(test_st) ptr_map;
+    static const size_t N             = 10 * (1 << 10);
```
Comment (yosefe) [inline test/gtest/ucs/test_datatype.cc:1164]: what is "N"? can use a more verbose name?

**Review by yosefe — APPROVED** [review]

**Comment by yosefe** [comment]
@evgeny-leksikov ok to merge?

**Review by evgeny-leksikov — APPROVED** [review]

**Pull request merged**